specviz/plugins/statistics/statistics_widget.py

```diff
@@ -222,12 +303,19 @@ def update_statistics(self):
         spectral_region = self._get_workspace_region()
 
         self._current_spectrum = spec
+        self._reconnect_item_signals()
```

Comment:
I don't think you should be running this test every time the update statistics is called. Why not have the plugin listen for selection changes in the workspace, and update the current item reference there?

```python
# In your __init__
self.hub.workspace.current_selection_changed.connect(lambda plot_item: self.update_statistics())
```

The signal returns the currently selected item, which could be passed to the update statistics function.

Reply:
We would still need a way to listen to unit changes. The main reason for this line is to make sure the stats widget updates when a unit changes. This is done by listening to `PlotDataItem.data_unit_changed` and `PlotDataItem.data_unit_changed`. I agree this is not the best way of doing it but there isn't a single signal for unit changes yet. Let me know if there is a better way to do this?